cfpb/cfgov-refresh
# Add ability to unit test Jinja2 macros with JSON specification

This PR adds utilities for unit testing Jinja2 macros based on JSON specifications and includes an example specification in `tests/`. The README should explain everything, if it doesn't I should change it so that it does.
 
**Additions**
 
- An abstract `MacroTestCase` Python class
- A factory function to take a test specified in a JSON file and create the appropriate Python class
- A test runner for JSON specification files
- A README that attempts to explain the process.
  
**Testing**
 
From within the `_tests/macro_testing` directory:

- Install the requirements (which should already be installed in a sheer/cfgov-refresh virtualenv): `pip install -r requirements.txt`
- Run the test runner: `python runner.py`
 
**Review**
 
- @anselmbradford 
- @sebworks 
- @jimmynotjim 

**Notes**

I presume that the new flapjack branch is the correct place for this.
